docs/content/docs/core-concept/transform.md

```diff
@@ -28,5 +28,245 @@ under the License.
+
+## Add filter rule
+Use reference columns when adding filtering rules to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    filter: id > 10 AND order_id > 100
+    description: project fields from source table
+```
+
+Use computed columns when adding filtering rules to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(province) as new_province 
+    filter: new_province = 'SHANGHAI'
+    description: project fields from source table
+```
+
+## Reassign primary key
+If reassign primary key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id
+    primary-keys: order_id
+    description: project fields from source table
+```
+
+If reassign composite primary key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(product_name) as product_name
+    primary-keys: order_id,product_name
+    description: project fields from source table
+```
+
+## Reassign partition key
+If reassign partition key to the table `web_order` in the database `mydb`, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(product_name) as product_name
+    partition-keys: product_name
+    description: project fields from source table
+```
+
+## Specify table creation configuration
+If specify the options to the table `web_order` in the database `mydb` when auto create table, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id, UPPER(product_name) as product_name
+    table-options: comment=web order
+    description: project fields from source table
+```
+Tips: The format of table-options is `key1=value1,key2=value2`.
+
+## Classification mapping
+
+Perform different processing on different classification data within a table, we can use this yaml file to define this transform：
+
+```yaml
+transform:
+  - source-table: mydb.web_order
+    projection: id, order_id
+    filter: UPPER(province) = 'SHANGHAI'
+    description: project fields from source table
+  - source-table: mydb.web_order
+    projection: order_id as id, id as order_id
+    filter: UPPER(province) = 'BEIJING'
+    description: project fields from source table
+```
+
+# Problem
+· Transform cannot be used with route in the current version, future versions will support it.
+· The computed column does not support referencing the trimmed original column in the current version, future versions will support it.
```

Comment:
```suggestion
* Computed columns cannot reference trimmed columns that do not present in final projection results. This will be fixed in future versions.
```